torch/csrc/distributed/c10d/init.cpp

```diff
@@ -1098,27 +1104,54 @@ that adds a prefix to each key inserted to the store.
   shared_ptr_class_<::gloo::transport::Device>(processGroupGloo, "Device");
 
   intrusive_ptr_class_<::c10d::ProcessGroupGloo::Options>(
-      processGroupGloo, "Options", processGroupOptions)
-      .def(py::init<>())
+      processGroupGloo,
+      "Options",
+      processGroupOptions,
+      R"(
+ProcessGroup options for Gloo backend
+
+Arguments:
+    timeout (timedelta, optional): Timeout for operations executed against
+            the process group. Default value equals 30 minutes.
+
+Example::
+    >>> import torch.distributed as dist
+    >>> import tempfile
+    >>> from datetime import timedelta
+    >>>
+    >>> temp_name = tempfile.NamedTemporaryFile(delete=False).name
+    >>> gloo_options = dist.ProcessGroupGloo.Options(timeout=timedelta(seconds=300))
+    >>> # initialize a gloo process group with the options just created
+    >>> dist.init_process_group("gloo", init_method=f"file://{temp_name}",
+    >>>                          world_size=2, rank=0, pg_options=gloo_options)
```

Comment:
Can we test the functionality when timeout is set in options and provided as an arg?